openapi/dcos.yaml

```diff
@@ -3877,6 +3998,330 @@ components:
+                not have to be integer, but can be a fraction.
+              minimum: 0.01
+            gpus:
+              type: integer
+              description: >-
+                The number of GPU shares this job needs per instance. This number does
+                not have to be integer, but can be a fraction.
+              minimum: 0
+            disk:
+              type: number
+              description: >-
+                How much disk space is needed for this job. This number does not have
+                to be an integer, but can be a fraction.
+              minimum: 0
+            ucr:
+              type: object
+              additionalProperties: false
+              nullable: true
+              properties:
+                image:
+                  type: object
+                  additionalProperties: false
+                  properties:
+                    id:
+                      type: string
+                      description: The repository image name.
+                    kind:
+                      type: string
+                      description: The type of image
+                      enum:
+                        - docker
+                        - appc
+                      default: docker
+                    forcePull:
+                      type: boolean
+                      description: >-
+                        The containerizer will pull the image from the registry, even
+                        if the image is already downloaded on the agent.
+                  required:
+                    - id
+                privileged:
+                  type: boolean
+                  description: Run this docker image in privileged mode
+              required:
+                - image
+            docker:
+              type: object
+              additionalProperties: false
+              nullable: true
+              properties:
+                image:
+                  type: string
+                  description: The docker repository image name.
+                forcePullImage:
+                  type: boolean
+                  description: >-
+                    The containerizer will pull the image from the registry, even if
+                    the image is already downloaded on the agent.
+                privileged:
+                  type: boolean
+                  description: Run this docker image in privileged mode
+                parameters:
+                  type: array
+                  description: ''
+                  items:
+                    type: object
+                    additionalProperties: false
+                    properties:
+                      key:
+                        type: string
+                      value:
+                        type: string
+                    required:
+                      - key
+                      - value
+              required:
+                - image
+            env:
+              type: object
```

Comment:
@bamarni any suggestion how to optimize this?

it somehow translates to `map[string]map[string]interface{}` but IMO it should be `map[string]interface{}` where the value could be asserted into string or map[string]string

Reply:
Agreed, what is generated is not optimal, I saw the same thing with Cosmos (https://github.com/dcos/client-go/blob/master/examples/cosmos/main.go#L99), but luckily there are always nested config properties there so I was fine with this.

I think this is the kind of edge case where we should report it upstream, in the meantime temporarily overwrite the generated code with a hardcoded template, until there is a proper solution.

Reply:
ok i'll check the openapi docs. maybe there is a better way defining this. I simply translated it from json schema which could also be the root cause for this